src/execution/riscv64/simulator-riscv64.cc

```diff
@@ -3121,33 +3121,45 @@ void Simulator::DecodeCRType() {
 }
 
 void Simulator::DecodeCAType() {
-  switch (instr_.RvcFunct6Value()) {
-    case 0b100011:
-      switch (instr_.RvcFunctValue()) {
-       case 0b00: //c.sub
-         if (instr_.RvcRdValue() != 0 && instr_.RvcRs2Value() != 0)
-           set_rvc_rs1s(sext_xlen(rvc_rs1s() - rvc_rs2s()));
-         else
-           UNSUPPORTED_RISCV();
-         break;
-       default:
-         UNSUPPORTED_RISCV();
-      }
-      break;
-    case 0b100111:
-      switch (instr_.RvcFunctValue()) {
-       case 0b01: //c.addw
-         if (instr_.RvcRdValue() != 0 && instr_.RvcRs2Value() != 0)
-           set_rvc_rs1s(sext32(rvc_rs1s() + rvc_rs2s()));
-         else
-           UNSUPPORTED_RISCV();
-         break;
-       default:
-         UNSUPPORTED_RISCV();
-      }
+  switch ((C1 | (instr_.InstructionBits() & kCATypeMask) |
+          (instr_.InstructionBits() & kRvcFunct2Mask))) {
+    case RO_C_SUB:
+      if (instr_.RvcRdValue() != 0 && instr_.RvcRs2Value() != 0)
+        set_rvc_rs1s(sext_xlen(rvc_rs1s() - rvc_rs2s()));
+      else
+        UNSUPPORTED_RISCV();
```

Comment:
The same as above,

1. Simplify the switch condition.
2. Remove if condition.

Besides, `UNSUPPORTED_RISCV();`is used in disasm, in simulator, it should be `UNSUPPORTED()`.